devenv/kustomize/e2e-tester.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: app-routing-operator-e2e
+  namespace: kube-system
+spec:
+  template:
+    metadata:
+      labels:
+        app: app-routing-operator-e2e
+    spec:
+      containers:
+        - name: tester
+          image: placeholderfortesterimage
+          command: [ "/go/e2e.test" ]
```

Comment:
this should just do `./e2e.test` and the Dockerfile just builds to ./e2e.test instead of /go/e2e.test. Having it be in the go directory breaks go conventions. I'd also be okay making this go/bin/e2e.test

Reply:
per our discussion resolving for future pr